packages/block-editor/src/store/selectors.js

```diff
@@ -1582,6 +1584,11 @@ const canInsertBlockTypeUnmemoized = (
 		return false;
 	}
 
+	const _isSectionBlock = !! isSectionBlock( state, rootClientId );
+	if ( _isSectionBlock ) {
```

Comment:
on scanning I thought the variable name would store whether the `clientId` is the section block. I think it might be worth changing the naming to clarify:
```suggestion
	const hasRootSectionBlock = !! isSectionBlock( state, rootClientId );
	if ( hasRootSectionBlock ) {
```

This does seem very similar to the `isBlockWithinSection` check (both check if parents are sections), so feels a bit duplicative. Maybe there's a good reason though (optimization?)

Reply:
It doesn't check the parent block, it check the rootClientId that gets passed. In other words, you can't insert blocks within the clientId that was passed. so it's whether the clientId is a section, kind of similar to the isLocked check right above.